### PR TITLE
Create ISSUE_TEMPLATE.md

### PR DESCRIPTION
**Pull request by nfelt**
Based loosely on TensorFlow's, but a little shorter and sweeter since we're a much smaller project.

### CODE REVIEW

File: ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Thank you in advance for reporting a TensorBoard bug, feature request, or documentation issue.  For more general support questions, please ask on Stack Overflow where a larger community can can help:
```
Comment (jart) [inline ISSUE_TEMPLATE.md]: Here's some wording inspired by stuff Martin says:

    The best place to get support is Stack Overflow.  
    They have a larger community with better searchability.

File: ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+https://stackoverflow.com/questions/tagged/tensorboard
+
+For bug reports, please include the following:
```
Comment (jart) [inline ISSUE_TEMPLATE.md]: `For bugs and feature requests, ...`

File: ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+ - TensorBoard version (from pip package, also printed out when running `tensorboard`)
+ - TensorFlow version if different from TensorBoard
+ - OS Platform and version (e.g., Linux Ubuntu 16.04)
+ - Python version (e.g. 2.7, 3.5)
+ - For browser-related issues:
+   - Browser type and version (e.g. Chrome 64.0.3282.140)
+   - Screenshot if it's a visual issue
+ 
+Please describe the bug as clearly as possible, and if possible provide a minimal example (code, data, and/or command line) to reproduce the issue.  Thanks!
```
Comment (jart) [inline ISSUE_TEMPLATE.md]: This line shouldn't be necessary.

File: ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+Thank you in advance for reporting a TensorBoard bug, feature request, or documentation issue.  For more general support questions, please ask on Stack Overflow where a larger community can can help:
+
+https://stackoverflow.com/questions/tagged/tensorboard
+
+For bug reports, please include the following:
+
+ - TensorBoard version (from pip package, also printed out when running `tensorboard`)
+ - TensorFlow version if different from TensorBoard
+ - OS Platform and version (e.g., Linux Ubuntu 16.04)
+ - Python version (e.g. 2.7, 3.5)
```
Comment (chihuahua) [inline ISSUE_TEMPLATE.md]: I like how this list includes examples.

File: ISSUE_TEMPLATE.md (hunk 2 above)
Comment (chihuahua) [inline ISSUE_TEMPLATE.md]: Maybe also include:

- Any logs or full tracebacks that would help diagnose the problem.
- Bazel version (if compiling from source)

File: ISSUE_TEMPLATE.md (hunk 3 above)
Comment (chihuahua) [reply ISSUE_TEMPLATE.md]: FWIW, here's what TensorFlow uses:

Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

Hence, maybe we could omit "describe the bug as clearly as possible"?

File: ISSUE_TEMPLATE.md (hunk 1 above)
Comment (nfelt) [reply ISSUE_TEMPLATE.md]: This seems pretty similar to what I have now?  I could reword it to something like: "For support, please use Stack Overflow.  They have a larger community with better searchability."

I'd like to stick with a slightly firmer "please use SO" versus just "SO is the best place" since I think otherwise people will tend to just ignore it.

File: ISSUE_TEMPLATE.md (hunk 2 above)
Comment (nfelt) [reply ISSUE_TEMPLATE.md]: I specifically left this as just bug reports, since this data is totally irrelevant for most feature requests we get, like "make log scale better" or "show line of code in the graph node info pane".  Mostly I just wanted this information for trying to reproduce bugs, and for feature requests there usually isn't anything to reproduce.  Thoughts?

Comment (nfelt) [reply ISSUE_TEMPLATE.md]: I'd rather put the request for logs/tracebacks with the more open-ended line at the bottom since it won't be relevant for all bugs.  Re: bazel version, I debated that but omitted it to keep the list short and sweet, since very few of our users seem to build TB from source, and most bugs are unlikely to depend on the bazel version unless they are build errors, at which point we might want to ask for more build environment specific info anyway.  Thoughts?

File: ISSUE_TEMPLATE.md (hunk 3 above)
Comment (nfelt) [reply ISSUE_TEMPLATE.md]: I think this line is useful... I wouldn't want somebody to include all the info above and then just say "the graphs aren't there" as their bug report.  As Chi pointed out, TF has a similar line, I just tried to shorten it a little.

If you like I can omit the "describe the bug as clearly as possible" or otherwise change the wording but I'd like to have at least something that reminds users to include 1) a minimal reproduction and 2) supporting artifacts (data / logs / tracebacks / etc.).  I could also fold this into the list above if we wanted.

File: ISSUE_TEMPLATE.md (hunk 2 above)
Comment (chihuahua) [reply ISSUE_TEMPLATE.md]: That makes sense - the majority of TensorBoard's users install via pip instead of building from the repo. And they'll probably mention it if they built with bazel since pip version wouldn't have meaning.

File: ISSUE_TEMPLATE.md (hunk 1 above)
Comment (chihuahua) [reply ISSUE_TEMPLATE.md]: I can see that, ie we want to prevent folks from seeking support via issues instead of just recommending SO.

File: ISSUE_TEMPLATE.md (hunk 2 above)
Comment (chihuahua) [reply ISSUE_TEMPLATE.md]: Indeed, feature requests don't follow this format. I wonder if it's worth indicating that issues are still for feature requests though. TF has this intro:

```
If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).
```

Maybe we can start with something like

```
GitHub issues should be bug reports or feature requests. For bug reports, please include the following:
```

Comment (nfelt) [reply ISSUE_TEMPLATE.md]: Well, this is why I had started out with "Thank you in advance for reporting a TensorBoard bug, feature request, or documentation issue."

I intended that to convey (a little more gently than TF's policy statement) that issues should be used for bugs, FRs, or doc issues.

Comment (chihuahua) [reply ISSUE_TEMPLATE.md]: Ah, I see!

**Comment by nfelt** [comment]
PTAL @jart

**Comment by nfelt** [comment]
@jart, friendly ping

File: ISSUE_TEMPLATE.md (hunk 1 above)
Comment (jart) [reply ISSUE_TEMPLATE.md]: Just for clarification, some folks consider "Thank you in advance" suboptimal, usually favoring, "Thanks for considering".

File: ISSUE_TEMPLATE.md (hunk 2 above)
Comment (jart) [reply ISSUE_TEMPLATE.md]: Please note we're going from empty to non-empty with this change. If in doubt, less might be more.
